File: gachisa-backend/src/main/java/com/gachisa/payment/entity/TossWebhookEvent.java
package com.gachisa.payment.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "toss_webhook_event")
public class TossWebhookEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String transmissionId;

    @Column(nullable = false)
    private String eventType;

    @Column(nullable = false)
    private String paymentKey;

    @Column(nullable = false)
    private LocalDateTime receivedAt;

    @Builder
    private TossWebhookEvent(String transmissionId, String eventType, String paymentKey,
                             LocalDateTime receivedAt) {
        this.transmissionId = transmissionId;
        this.eventType = eventType;
        this.paymentKey = paymentKey;
        this.receivedAt = receivedAt;
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/entity/TossWebhookEvent.kt
package com.gachisa.payment.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "toss_webhook_event")
class TossWebhookEvent(
    @Column(nullable = false, unique = true)
    val transmissionId: String,

    @Column(nullable = false)
    val eventType: String,

    @Column(nullable = false)
    val paymentKey: String,

    @Column(nullable = false)
    val receivedAt: LocalDateTime,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
        protected set
}
